src/compiler/checker.ts

```diff
@@ -20306,16 +20313,23 @@ namespace ts {
                 const propDeclaration = prop.valueDeclaration;
 
                 // index is numeric and property name is not valid numeric literal
-                if (indexKind === IndexKind.Number && !(propDeclaration ? isNumericName(propDeclaration.name) : isNumericLiteralName(prop.name))) {
+                if (indexKind === IndexKind.Number && !(propDeclaration ? isNumericName(getNameOfDeclaration(propDeclaration)) : isNumericLiteralName(prop.name))) {
                     return;
                 }
 
+                let errorNode: Node;
+                if (propDeclaration && propDeclaration.kind === SyntaxKind.BinaryExpression) {

```

Comment:
yeah, good point. It's unlikely that any other declaration (even a special ModuleExport) would get in here by mistake.